providers-sdk/v1/recording/recording.go

```diff
@@ -355,6 +357,10 @@ func (r *recording) findAssetConnID(asset *inventory.Asset) (int, string) {
 func (r *recording) EnsureAsset(asset *inventory.Asset, providerID string, connectionID uint32, conf *inventory.Config) {
 	found, _ := r.findAssetConnID(asset)
 
+	if asset.Platform == nil {
```

Comment:
one example is connecting to a cloud provider: we return an 'empty' asset initially from discovery which has nothing set. this ensures we do not break the scan/shell

a better fix here would be to ensure we return only the right assets from discovery but this is still a nice safeguard